Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -1068,4 +1065,240 @@ function Set-ClusterDefaultStoragePolicy {
     }
 }
 
-Export-ModuleMember -Function *
+<#
+    .Synopsis
+    Creates a temporary user and a role which includes required privileges.
+    The user is used to pass credentials into vcenter rest api calls for functionalites unavailable to powercli
+
+    .Parameter privileges
+    Array of user privileges (default value is an empty array)
+
+    .Parameter userName
+    User-Friendly name for the user
+
+    .Parameter userRole
+    User-Friendly name for the role
+
+    .Example
+    New-TempUser -privileges @("VirtualMachine.Config.CPUCount","VirtualMachine.Config.Memory") -userName TempUser -userRole TempRole
+#>
+function New-TempUser {
+    Param (
+        [Parameter(
+            Mandatory = $false,
+            HelpMessage = "Array of user privileges")]
+        [array]
+        $privileges = @(),
+
+        [Parameter(
+            Mandatory = $true,
+            HelpMessage = 'User-Friendly name for the user')]
+        [ValidateNotNull()]
+        [string]
+        $userName,
+
+        [Parameter(
+            Mandatory = $true,
+            HelpMessage = 'User-Friendly name for the role')]
+        [ValidateNotNull()]
+        [string]
+        $userRole
+    )
+
+    if($privileges.Count -eq 0) { throw "If adding a temporary user, please ensure you pass in at least one privilege"}
+
+    $domain = "vsphere.local"
+    $group = "CloudAdmins"
```

Comment:
This group memembership probably should be optional - it has prohibitions that would prevent access to mgmt resources (like HCX appliance).

Reply:
Update the params to include the group of a non-mandatory parameter with the default value as "CloudAdmins"